internal/host/plugin/repository_host_catalog_test.go

```diff
@@ -293,10 +320,26 @@ func TestRepository_CreateCatalog(t *testing.T) {
 			assert.Equal(tt.want.Description, got.Description)
 			assert.Equal(got.CreateTime, got.UpdateTime)
 
+			if origPluginAttrs != nil {
```

Comment:
These tests are failing for me locally. The `main` branch tests pass for me. Here is the error:

```
--- FAIL: TestRepository_CreateCatalog (1.09s)
    --- FAIL: TestRepository_CreateCatalog/valid-with-attributes (0.02s)
        repository_host_catalog_test.go:343:
                Error Trace:    /Users/mgaffney/sandbox/go/gomod/boundary-review/internal/host/plugin/repository_host_catalog_test.go:343
                Error:          Not equal:
                                expected: []byte{0xa, 0x27, 0xa, 0x12, 0x6e, 0x6f, 0x72, 0x6d, 0x61, 0x6c, 0x69, 0x7a, 0x65, 0x5f, 0x74, 0x6f, 0x5f, 0x73, 0x6c, 0x69, 0x63, 0x65, 0x12, 0x11, 0x32, 0xf, 0xa, 0xd, 0x1a, 0xb, 0x6e, 0x6f, 0x72, 0x6d, 0x61, 0x6c, 0x69, 0x7a, 0x65, 0x6d, 0x65, 0xa, 0xb, 0xa, 0x2, 0x6b, 0x31, 0x12, 0x5, 0x1a, 0x3, 0x66, 0x6f, 0x6f}
                                actual  : []byte{0xa, 0xb, 0xa, 0x2, 0x6b, 0x31, 0x12, 0x5, 0x1a, 0x3, 0x66, 0x6f, 0x6f, 0xa, 0x27, 0xa, 0x12, 0x6e, 0x6f, 0x72, 0x6d, 0x61, 0x6c, 0x69, 0x7a, 0x65, 0x5f, 0x74, 0x6f, 0x5f, 0x73, 0x6c, 0x69, 0x63, 0x65, 0x12, 0x11, 0x32, 0xf, 0xa, 0xd, 0x1a, 0xb, 0x6e, 0x6f, 0x72, 0x6d, 0x61, 0x6c, 0x69, 0x7a, 0x65, 0x6d, 0x65}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,6 +1,6 @@
                                 ([]uint8) (len=54) {
                                - 00000000  0a 27 0a 12 6e 6f 72 6d  61 6c 69 7a 65 5f 74 6f  |.'..normalize_to|
                                - 00000010  5f 73 6c 69 63 65 12 11  32 0f 0a 0d 1a 0b 6e 6f  |_slice..2.....no|
                                - 00000020  72 6d 61 6c 69 7a 65 6d  65 0a 0b 0a 02 6b 31 12  |rmalizeme....k1.|
                                - 00000030  05 1a 03 66 6f 6f                                 |...foo|
                                + 00000000  0a 0b 0a 02 6b 31 12 05  1a 03 66 6f 6f 0a 27 0a  |....k1....foo.'.|
                                + 00000010  12 6e 6f 72 6d 61 6c 69  7a 65 5f 74 6f 5f 73 6c  |.normalize_to_sl|
                                + 00000020  69 63 65 12 11 32 0f 0a  0d 1a 0b 6e 6f 72 6d 61  |ice..2.....norma|
                                + 00000030  6c 69 7a 65 6d 65                                 |lizeme|
                                 }
                Test:           TestRepository_CreateCatalog/valid-with-attributes
```

Reply:
Ugh, I had such weird issues with that one yesterday and restarting vscode fixed it so I thought it was some cached environment thing. I'll take another look.

The actual data is fine as far as I can tell, it's that it's marshaling differently for some reason.

Reply:
Protobuf marshalling is not deterministic 😬. The best test here may be to unmarshal and check values rather than byte by byte comparisons.

Reply:
Oh! The test already used byte by byte comparisons -- I'm just modifying what is in the bytes -- but I didn't know it's not deterministic. I thought it was, which was why I thought the problem lay with something weird I was doing rather than the existing test.

Reply:
Fixed.